### ISSUE TITLE
SDK stops working if NET6.0 (preview) is installed 

### ISSUE THREAD

**Issue by pinkfloydx33** [rr-wfm/MSBuild.Sdk.SqlProj#165]
As the title says, this project does not work if NET6.0 SDK is installed, even if side-by-side with NET5.0 (which is supported) or earlier supported frameworks. This is due to the checks here:

https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/blob/4d872dd45edcca5a741ae6855d81276ff25df3ad/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets#L92-L98

I was able to side-step the issue by adding a global.json file to my repo that capped the SDK to the 5.0.x line, however I'm wondering if there's something we can do here. I'm not asking for 6.0 support since it's still in preview, but I'm wondering if there's some way for this SDK to force itself to a *supported, installed* framework. I don't know the particulars of the MSBuild/SDK system so this may not be possible.... But it was quite confusing when my build started failing simply because I installed the preview SDK on my machine (side-by-side with 5.0).

A related issue that I noticed due to all of this is the error message on line 98 which says this sdk only supports 2.1/3.1:
https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/blob/4d872dd45edcca5a741ae6855d81276ff25df3ad/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets#L98-L98

This SDK does support NET5.0...it seems like the error message was missed during that update.

**Comment by ErikEJ** [comment]
Also the link in the message needs updated

**Comment by jmezach** [comment]
I'll have a look at updating the link in the message when I have some time.

Unfortunately I don't think there's much we can do in terms of forcing the use of the non-preview SDK. We determine the runtime version that comes bundled with the SDK since we need the runtime in order to run the command line tool. It is my understanding that as soon as you install a preview version of the .NET SDK you'll use that version and the only thing you can do to change that is by using the `global.json` as you've already mentioned.

I'm planning on supporting .NET 6 as soon as it has a Go-Live license as I'm guessing more users will be using it then.

**Comment by pinkfloydx33** [comment]
Yah I figured as much. I have the workaround so not a big deal. Thanks. 

Would it make sense to update the error message to include something like "or add a global.json file...."? I haven't needed to use such a file since the early days of netcore and forgot they even existed! Would have saved me a little bit of time, perhaps it might help someone else.

**Comment by jmezach** [comment]
Yeah, I think that makes sense to point people at the global.json file. Feel free to open a PR for updating the message here. Not sure when I'll have time to do it myself ;).

**Comment by jmezach** [comment]
The improved error message has finally shipped in 1.16.2. @ErikEJ Thanks for the contribution!

**Comment by Condor2708** [comment]
Can someone share an example how to configure the global.json file?

**Comment by eiki25** [comment]
@jmezach, @ErikEJ any news on this issue?
Now that net6.0 is not a preview anymore, and there is no "**netcoreapp**6.0" TargetFramework.

Creating global.json with lower framework version is not a good workaround because it makes it impossible to have .NET 6 projects along with MsBuild Sdk SqlProjects.

**Comment by jmezach** [comment]
Unfortunately it turned out that a library we're using to build the `.dacpac` that's provided by Microsoft (DACFX) didn't support .NET 6. That has been fixed, but hasn't yet been released so we're still waiting on that before we can ship a new version of MSBuild.Sdk.SqlProj that can support .NET 6.

**Comment by ErikEJ** [comment]
Could we ship a preview?

**Comment by jmezach** [comment]
Yeah, maybe, although I still had another issue to tackle in #194

**Comment by Condor2708** [comment]
> @jmezach, @ErikEJ any news on this issue? Now that net6.0 is not a preview anymore, and there is no "**netcoreapp**6.0" TargetFramework.
> 
> Creating global.json with lower framework version is not a good workaround because it makes it impossible to have .NET 6 projects along with MsBuild Sdk SqlProjects.

Another workaround is to have installed both SDKs for 5.0 and 6.0 and set this variable into the csproj 
    <BundledNETCoreAppTargetFrameworkVersion>5.0</BundledNETCoreAppTargetFrameworkVersion>
